src/bun.js/bindings/JSBuffer.cpp

```diff
@@ -1068,17 +1068,72 @@ static inline JSC::EncodedJSValue jsBufferPrototypeFunction_lastIndexOfBody(JSC:
 static inline JSC::EncodedJSValue jsBufferPrototypeFunction_swap16Body(JSC::JSGlobalObject* lexicalGlobalObject, JSC::CallFrame* callFrame, typename IDLOperation<JSBuffer>::ClassParameter castedThis)
 {
     auto& vm = JSC::getVM(lexicalGlobalObject);
-    return JSC::JSValue::encode(jsUndefined());
+    auto scope = DECLARE_THROW_SCOPE(vm);
+
+    const int size = 2;
+    int64_t length = static_cast<int64_t>(castedThis->byteLength());
+    if (length % size != 0) {
+        throwRangeError(lexicalGlobalObject, scope, "Invalid buffer length"_s);
+        return JSC::JSValue::encode(jsUndefined());
+    }
+
+    uint8_t* typedVector = castedThis->typedVector();
```

Comment:
Typed Arrays can become "detached". When detached, the `vector` or `typedVector` will return a `nullptr`.  That means we have to check for this. It's fortunately pretty uncommon, so we can mark the branch as `UNLIKELY`

```c++
    if (UNLIKELY(castedThis->isDetached())) {
        throwVMTypeError(lexicalGlobalObject, throwScope, "Buffer is detached"_s);
        return JSValue::encode(jsUndefined());
    }
```